### PR TITLE
Add get styles

### PR DESCRIPTION
**Pull request by Vampouille**
This is same PR as #1030

### CODE REVIEW

**Comment by Vampouille** [comment]
In order to merge this PR, I should allow usage of getStyles query for all WMS version, not only 1.1.1 ?

**Comment by aaime** [comment]
I believe that's what @jodygarnett wanted. Jody, care to confirm?

**Comment by Vampouille** [comment]
Here is version that work with any WMS version. I also fix some mistake in javadoc.

**Comment by Vampouille** [comment]
@jodygarnett can you take a look at my last commit ? is there anything I can add ?

**Comment by aaime** [comment]
@jodygarnett do you have time to look at this one?

**Comment by Vampouille** [comment]
Is it possible to merge this PR before code freeze. It would be nice to have this improvement in new version.

**Comment by bencaradocdavies** [comment]
@jodygarnett you reviewed the predecessor https://github.com/geotools/geotools/pull/1030 of this pull request. Does this new pull request have your approval?

**Comment by Vampouille** [comment]
Hello, 
Seems like @jodygarnett is not available for now. Is there another committer willing to review this PR ? Maybe @groldan or @jericks ?

File: modules/extension/wms/pom.xml [hunk 1]
```diff
@@ -143,10 +143,22 @@
...
+    <dependency>
     	<groupId>org.mockito</groupId>
     	<artifactId>mockito-all</artifactId>
     	<scope>test</scope>
     </dependency>
+    <dependency>
+	    <groupId>javax.mail</groupId>
+	    <artifactId>mail</artifactId>
+	    <version>1.4.7</version>
+    </dependency>
```
Comment (bencaradocdavies) [inline modules/extension/wms/pom.xml]: @Vampouille surely this should only be a test scope dependency? Also, this change introduces new tabs. Place use spaces for indentation.

File: modules/extension/wms/src/main/java/org/geotools/data/wms/WMS1_0_0.java [hunk 2]
```diff
@@ -25,19 +25,11 @@
...
-import org.geotools.data.ows.HTTPResponse;
-import org.geotools.data.ows.Response;
-import org.geotools.data.wms.request.AbstractGetFeatureInfoRequest;
-import org.geotools.data.wms.request.AbstractGetMapRequest;
-import org.geotools.data.wms.request.DescribeLayerRequest;
-import org.geotools.data.wms.request.GetLegendGraphicRequest;
-import org.geotools.data.wms.request.GetStylesRequest;
-import org.geotools.data.wms.request.PutStylesRequest;
+import org.geotools.data.ows.*;
+import org.geotools.data.wms.request.*;
```
Comment (bencaradocdavies) [inline modules/extension/wms/src/main/java/org/geotools/data/wms/WMS1_0_0.java]: @Vampouille I frown on wildcard imports, even though our coding conventions are silent on the matter. The Eclipse default is 99 imports before a package wildcard is used. Was this change intended? What editor are you using?

File: modules/extension/wms/src/main/java/org/geotools/data/wms/WMS1_1_1.java [hunk 3]
```diff
@@ -16,9 +16,20 @@
...
 
 import org.geotools.data.ows.GetCapabilitiesRequest;
+import org.geotools.data.ows.HTTPResponse;
+import org.geotools.data.ows.Response;
+import org.geotools.data.wms.request.AbstractGetStylesRequest;
+import org.geotools.data.wms.request.AbstractPutStylesRequest;
+import org.geotools.data.wms.request.GetStylesRequest;
+import org.geotools.data.wms.request.PutStylesRequest;
+import org.geotools.data.wms.response.GetStylesResponse;
+import org.geotools.data.wms.response.PutStylesResponse;
```
Comment (bencaradocdavies) [inline modules/extension/wms/src/main/java/org/geotools/data/wms/WMS1_1_1.java]: @Vampouille these imports are fine, but inconsistent with the wildcards you introduced elsewhere!

File: modules/extension/wms/src/test/java/org/geotools/data/wms/test/LocalGeoServerOnlineTest.java [hunk 4]
```diff
@@ -32,16 +36,15 @@
 import org.apache.commons.io.output.StringBuilderWriter;
 import org.geotools.data.ResourceInfo;
 import org.geotools.data.ServiceInfo;
-import org.geotools.data.ows.CRSEnvelope;
-import org.geotools.data.ows.Layer;
-import org.geotools.data.ows.MultithreadedHttpClient;
-import org.geotools.data.ows.OperationType;
-import org.geotools.data.ows.WMSCapabilities;
+import org.geotools.data.ows.*;
```
Comment (bencaradocdavies) [inline modules/extension/wms/src/test/java/org/geotools/data/wms/test/LocalGeoServerOnlineTest.java]: More new wildcard imports.

File: modules/extension/wms/src/test/java/org/geotools/data/wms/test/LocalGeoServerOnlineTest.java [hunk 5]
```diff
@@ -50,6 +53,9 @@
 import org.geotools.referencing.CRS.AxisOrder;
 import org.geotools.referencing.crs.DefaultEngineeringCRS;
 import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.geotools.styling.*;
+import org.geotools.styling.builder.*;
```
Comment (bencaradocdavies) [inline modules/extension/wms/src/test/java/org/geotools/data/wms/test/LocalGeoServerOnlineTest.java]: Wildcard imports.

File: modules/extension/wms/src/main/java/org/geotools/data/wms/WMS1_0_0.java (hunk 2 above)
Comment (Vampouille) [inline modules/extension/wms/src/main/java/org/geotools/data/wms/WMS1_0_0.java]: I'm using IntelliJ, I agree with you about wildcard, it reduce readability. So turn this feature off.

**Comment by Vampouille** [comment]
Thanks for review, this last commit should fix package imports and unwanted tabs.

**Comment by bencaradocdavies** [comment]
Thanks very much, @Vampouille. I also created a Jira issue so that your improvement is mentioned in the release notes:
https://osgeo-org.atlassian.net/browse/GEOT-5356
